ZCRMiOS/Classes/DashBoardAPIHandler.swift

```diff
@@ -0,0 +1,1737 @@
+                
+                let JSONlayoutValues =
+                    getDashBoardMetaComponentLayoutPropsFrom(metaComponentJSON)
+                
+                var metaComponentLayoutObj = metaComponentObj.getLayoutProperties()
+                
+                metaComponentLayoutObj.setComponentX(Position: JSONlayoutValues.xPosition)
+                metaComponentLayoutObj.setComponentY(Position: JSONlayoutValues.yPosition)
+                metaComponentLayoutObj.setComponent(Width: JSONlayoutValues.width)
+                metaComponentLayoutObj.setComponent(Height: JSONlayoutValues.height)
+                
+                metaComponentObj.setLayoutProperties(metaComponentLayoutObj)
+                
+                
+            default:
+                print("UNKNOWN KEY \(key) ENCOUNTERED IN DASHBOARD META COMPONENT PARSING !!")
+                
+            }
+            
+        }
+        
+        return metaComponentObj
+    }
+    
+    
+    
+    internal func getDashBoardMetaComponentLayoutPropsFrom(_ metaComponentJSON:[String:Any]) ->   MetaComponentLayoutPropsTuple {
+        
+        var width:Int?
+        var height:Int?
+        var xPosition:Int?
+        var yPosition:Int?
+        
+        
+        let itemPropsJSON =
+            metaComponentJSON.getDictionary(key: metaComponentAPINames.itemProps)
+        
+        let layoutJSON = itemPropsJSON.getDictionary(key: metaComponentAPINames.layout)
+        
+        
+        for (key,value) in layoutJSON {
+            
+            switch (key) {
+                
+            case metaComponentAPINames.componentWidth:
+                width = Int((value as? String) ?? "default")
+                
+            case metaComponentAPINames.componentHeight:
+                height = Int((value as? String) ?? "default")
+                
+            case metaComponentAPINames.componentXPosition:
+                xPosition = Int((value as? String) ?? "default")
+                
+            case metaComponentAPINames.componentYPosition:
+                yPosition = Int((value as? String) ?? "default")
+                
+            default:
+                print("UNKNOWN KEY \(key) ENCOUNTERED IN DASHBOARD META COMPONENT LAYOUT PARSING !!")
+                
+            }
+            
+        }
+        
+        return (width: width, height: height, xPosition: xPosition, yPosition: yPosition)
+    }
+    
+    
+} // Extension ends
+
+
+
+extension DashBoardAPIHandler {
+    
+    //MARK:- *** DASHBOARD COMPONENT PARSERS ***
+    
+    
+    internal func getDashBoardComponentFrom(_ componentJSON: [String:Any]) -> ZCRMDashBoardComponent {
+        
+        let componentObj = ZCRMDashBoardComponent()
+        
+        for(key,value) in componentJSON {
+            
+            switch(key) {
+                
+            case componentAPINames.componentName:
+                componentObj.setComponent(Name: value as? String)
```

Comment:
Why didn't you use our dictionary extension?
Remove empty lines

Reply:
https://github.com/zoho/CRM-iOSSDK/pull/15#discussion_r214515160